### PR TITLE
[AMS] Fix #14021: az ams account sp is not idempotent

### LINKED ISSUE #14021: az ams account sp is not idempotent
Consequently executing `az ams account sp create` fails with error `Service principal already exists.`. Same bug has been already [resolved ](https://github.com/Azure/azure-cli/pull/9108) for `az ad app create` but still a case for AMS.

### PR DESCRIPTION
**Pull request by hivyas**
**Description**  
Fix for #14021 

**Testing Guide**  
Running the following command multiple times should output the same SP object
`az ams sp create -n [sp-name] -a [account-name] -g [resource-group]`

To change the name or role of an existing SP run the following command:
`az ams sp create -n [sp-name] --new-sp-name [new-sp-name] -a [account-name] -g [resource-group] --role [role]`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help reviewing this PR? thanks

File: src/azure-cli/azure/cli/command_modules/ams/operations/sp.py [hunk 1]
```diff
@@ -290,12 +328,17 @@ def _assign_role(cmd, role, sp_oid, scope):
 
 
 def _build_sp_result(subscription_id, location, resource_group_name, account_name,
-                     tenant, app_id, sp_password, management_endpoint,
-                     active_directory_endpoint, resource_manager_endoint, xml):
+                     tenant, app_id, sp_name, sp_password, management_endpoint,
+                     active_directory_endpoint, resource_manager_endoint, role, xml):
+    if not sp_password:
+        sp_password = "Cannot redisplay resecret. Please use reset-credentials to generate a new secret."
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/ams/operations/sp.py:334]: Is `resecret` a typo?

File: src/azure-cli/azure/cli/command_modules/ams/commands.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def get_custom_sdk(custom_module, client_factory):
         g.command('sync-storage-keys', 'sync_storage_keys')
 
     with self.command_group('ams account sp', get_sdk('Mediaservices', get_mediaservices_client)) as g:
-        g.custom_command('create', 'create_assign_sp_to_mediaservice',
+        g.custom_command('create', 'create_or_update_assign_sp_to_mediaservice',
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/ams/commands.py:57]: Please also update the help text to reflect the change in `create` behavior:
https://github.com/Azure/azure-cli/blob/91528ee56c70018e5a2cb56cf6527fc9b2013879/src/azure-cli/azure/cli/command_modules/ams/_help.py#L70-L72

File: src/azure-cli/azure/cli/command_modules/ams/operations/sp.py (hunk 1 above)
Comment (hivyas) [reply src/azure-cli/azure/cli/command_modules/ams/operations/sp.py]: Yup, thanks for catching that

**Pull request merged**